Finish all initial features

chrissank opened the pull request:
All features are done, incl. unit tests

If anyone wants to review it, please do, this is pretty much the entire codebase being changed.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/minerva/7)
<!-- Reviewable:end -->

chrissank commented:
@jacobdeery would you like to take a look at the unit tests?

Oasixer commented on README.md:
```diff
@@ -6,47 +6,53 @@ Minerva is Waterloo Rocketry's soon-to-be slackbot. Minerva will handle all meet
 - `<link-to-message>`. The link to the message you want to copy. (hover on the message -> hit the dots -> copy link) 
+- `<copy/alert/alert-single-channel>`. One of these three parameters must be selected.
+    - `copy` Simply copies the link to each desired channel.
+    - `alert` Append an `@channel` infront of the linked message, then sends this updated message to each channel. **Note**: A 5-channel limit is imposed on this function to restrict channel-bombs.
```
```suggestion
    - `alert` Append a `@channel` infront of the linked message, then sends this updated message to each channel. **Note**: A 5-channel limit is imposed on this function to restrict channel-bombs.
```

Oasixer commented on README.md:
```diff
@@ -6,47 +6,53 @@ Minerva is Waterloo Rocketry's soon-to-be slackbot. Minerva will handle all meet
 #### Examples: 
 
 - Today is the first meeting of project proposals! Shirley wants everyone to alert everyone on the team, so she says: `@everyone project proposals are today at 8pm! Here is the link: https://example.com`. She wants to *alert* everyone of this message, so she uses `/notify <link-to-message> alert-single-channel`
 - It's the time of the project proposals meeting, but Shirley doesn't want to ping the single-channel guests again. Instead she just wants to post the meeting link accross all channels. She uses: `/notify <link-to-message>`
```
```suggestion
- It's the time of the project proposals meeting, but Shirley doesn't want to ping the single-channel guests again. Instead she just wants to post the meeting link across all channels. She uses: `/notify <link-to-message>`
```

Oasixer commented on functions/commands/notify.js:
```diff
@@ -4,17 +4,17 @@ module.exports.send = async function (user_id, textParams, initialChannel) {
             await slack_handler.directMessageSingleChannelGuestsInChannels(message + "\n\n_You have been sent this message because you are a single channel guest who might have otherwise missed this alert._", parameters.channels);
         } else {
-            // default is to just copy to all channels with message (that can contain alert or not)
+            // otherwise, just copy the message to the channes. It may have an 'alert' appended to it.
```
```suggestion
            // otherwise, just copy the message to the channels. It may have an 'alert' appended to it.
```

Oasixer commented:
I fixed a couple typos with a lil commit, but overall it's really nice, especially the readme and the unit tests. 9/10 would read again. Didn't find any problems.

chrissank commented:
>
> in front
> 
> Remove colon and trailing whitespace
> 
> Remove trailing space
> 
> it's
> 
> remove?
>
> just `return lines[2] === "default" || lines[1] === "alert-single-channel";`?
> 

Done

> 
> Is this the desired behaviour, or should there be parentheses around the `||` conditions?
>

That is the intended behaviour, I have added parenthesis for clarity.

> 
> "positive integer"? or are floats allowed?
> 

Positive integers is the intended function, changed error message to reflect this.

>
> Where does this actually get logged to, out of curiosity?
>

This gets logged to the Firebase Console at the moment, however, I will also mirror this log in #minerva-log 


> 
> what does this comment mean?
> 

Comment changed to better reflect what it does (it means that events can happen at the same time, so we want to pull next 4 events to be sure that we don't miss any at either 5 minutes or 6 hours from now)


> 
> Why this particular emoji?
> 

No particular reason. Currently this is a placeholder since getting a list of emojis is an slack enterprise locked feature. A curated list of emojis can also work.

> 
> Numbers like this would benefit from being pulled into named constants for clarity
> 
> Same here
> 
Changed to two constants

>
> * plain `remove` with nothing else
> * `remove 1.5` or another non-integer numeric
> 

These tests are captured in the `filterParameters`-`remove non-number` and  `filterParameters`-`remove number less than 1`

As for floats, `parseInt` will automatically cast floats to integers, so they should get handled nicely.

> 
> There are a lot of long single lines in these tests. Consider splitting them up to improve readability.
> 

👍
